src/Context/DataProvider.js

```diff
@@ -0,0 +1,51 @@
+import { useContext, useReducer, useState } from "react";
+import { DataContext } from "./DataContext";
+import { ReducerFunction } from "./ReducerFunction";
+import { data } from "./Data";
+export const DataProvider = ({ children }) => {
+  const [state, dispatch] = useReducer(ReducerFunction, {
+    videos: data,
+    categories: [],
+    playlists: [
+      {
+        id: 123,
+        name: "Watch Later",
+        videos: ["8egWflM1xok"],
```

Comment:
I was just trying to check the UI of playlists displaying parts like if a playlist has a video its image will be the image of the first video else it will be some random image. So I have added a default value.